gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +721,671 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+
+        fits_read_key(hFITS, TSTRING, "OBJECT", target, nullptr, &status);
+        if( status )
+        {
+            strncpy(target, "Undefined", 10);
+            CPLError(CE_Failure, CPLE_AppDefined,
+                 "OBJECT keyword is missing");
+            status = 0;
+        }
+
+        strncpy(pszGeogName, "GCS_",5);
+        strcat(pszGeogName, target);
+        strncpy(pszDatumName, "D_",3);
+        strcat(pszDatumName, target);
+
+        fits_read_key(hFITS, TDOUBLE, "A_RADIUS", &aRadius, nullptr, &status);
+        if( status )
+        {
+            CPLError(CE_Failure, CPLE_AppDefined,
+                 "No Radii keyword available, metadata will not contain DATUM information.");
+            status = 0;
+        } else {
+            fits_read_key(hFITS, TDOUBLE, "C_RADIUS", &cRadius, nullptr, &status);
+            if( aRadius != cRadius )
+            {
+                invFlattening = aRadius / ( aRadius - cRadius );
+            }
+        }
+
+        /* Waiting for linear keywords standardization only deg ctype are used */
+        /* Check if WCS are there */
+        fits_read_key(hFITS, TSTRING, "CTYPE1", ctype, nullptr, &status);
+        if ( !status ) {
+            /* Check if angular WCS are there */
+            if ( strstr(ctype, "LN") )
+            {
+                /* Reading reference points */
+                fits_read_key(hFITS, TDOUBLE, "CRPIX1", &crpix1, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRPIX2", &crpix2, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRVAL1", &crval1, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRVAL2", &crval2, nullptr, &status);
+                /* Check for CDELT and PC matrix representation */
+                fits_read_key(hFITS, TDOUBLE, "CDELT1", &cdelt1, nullptr, &status);
+                if ( ! status ) {
+                    fits_read_key(hFITS, TDOUBLE, "CDELT2", &cdelt2, nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC1_1", &pc[0], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC1_2", &pc[1], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC2_1", &pc[2], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC2_2", &pc[3], nullptr, &status);
+                    cd[0] = cdelt1 * pc[0];
+                    cd[1] = cdelt1 * pc[1];
+                    cd[2] = cdelt2 * pc[2];
+                    cd[3] = cdelt2 * pc[3];
+                    status = 0;
+                } else {
+                    /* Look for CD matrix representation */
+                    fits_read_key(hFITS, TDOUBLE, "CD1_1", &cd[0], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD1_2", &cd[1], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD2_1", &cd[2], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD2_2", &cd[3], nullptr, &status);
+                }
```

Comment:
Does `status` need to be set to 0 here like in the case above?

Reply:
@abellgithub you're right, my checks on `status` are a bit randomized... :(, working on improve them. Thanks.